### PR TITLE
[promtail] Adding serviceMonitor scheme and tlsConfig values

### PR DESCRIPTION
**Pull request by rgarcia6520**
Signed-off-by: garcia.ryan <garcia.ryan@solute.us>

Adding `scheme` & `tlsConfig` values to `serviceMonitor` for promtail.

### CODE REVIEW

**Comment by verejoel** [comment]
Didn't see this before I submitted #1918, really need this TLS config change in

File: charts/promtail/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -41,4 +41,11 @@ spec:
       metricRelabelings:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      {{- if .Values.serviceMonitor.scheme }}
+      scheme: {{ .Values.serviceMonitor.scheme }}
+      {{- end }}
+      {{- if .Values.serviceMonitor.tlsConfig }}
+      tlsConfig:
+        {{- toYaml .Values.serviceMonitor.tlsConfig | nindent 8 }}
+      {{- end }}
```
Comment (zanhsieh) [inline charts/promtail/templates/servicemonitor.yaml:50]:
```suggestion
      {{- with .Values.serviceMonitor.tlsConfig }}
      tlsConfig:
        {{- toYaml . | nindent 8 }}
      {{- end }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/promtail/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -41,4 +41,11 @@ spec:
...
 {{- end }}
```
[the remark this answers is not in the document]
Comment (verejoel) [reply charts/promtail/templates/servicemonitor.yaml:51]:
```suggestion
  {{- with .targetLabels }}
  targetLabels:
    {{- toYaml . | nindent 4 }}
  {{- end }}
{{- end }}
```

File: charts/promtail/values.yaml [hunk 3]
```diff
@@ -203,6 +203,10 @@ serviceMonitor:
   # https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#relabelconfig
   # (defines `metric_relabel_configs`)
   metricRelabelings: []
```
Comment (verejoel) [inline charts/promtail/values.yaml:205]:
```suggestion
  metricRelabelings: []
    # --ServiceMonitor will add labels from the service to the Prometheus metric
  # https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#servicemonitorspec
  targetLabels: []
```

**Comment by zanhsieh** [comment]
It seems this PR already superseded by other PRs. Close it for now.

**Pull request closed without merging**